### ISSUE TITLE
Export StarDist model, re-use in FIJI

### ISSUE THREAD

**Issue by romainGuiet** [mpicbg-csbd/stardist#10]
Hi StarDist Team,

I’m playing with StarDist for a couple of days now and it’s amazing !

I just trained a model with your datasets and the network settings from the notebook and use this model to make some predictions on my own images. The results is (as expected) sensitive to scaling but it amazingly good!

I’m now looking for a solution to use the model in FIJI.

I tried to model.exportTF() as in CSBDeep, but the export function is not implemented (yet ?).  
Will you release “soon” or do you plan to make a plugin for FIJI?

Thanks again for this amazing tool!
Cheers,
Romain

**Comment by uschmidt83** [comment]
Hi Romain,

> I’m playing with StarDist for a couple of days now and it’s amazing !

that's great to hear!

> I just trained a model with your datasets and the network settings from the notebook and use this model to make some predictions on my own images. The results is (as expected) sensitive to scaling but it amazingly good!

The [pre-trained model](https://github.com/mpicbg-csbd/stardist/tree/master/models/dsb2018) should actually work decently on a variety of fluorescence microscopy data.
What do you mean by "sensitive to scaling"?

> I’m now looking for a solution to use the model in FIJI.
> 
> I tried to model.exportTF() as in CSBDeep, but the export function is not implemented (yet ?).
> Will you release “soon” or do you plan to make a plugin for FIJI?

I have a basic StarDist Fiji plugin working, but can't promise at the moment when it'll be done.
However, there is the possibility to export the StarDist projections as ImageJ1 ROIs. The Python code is done, but I haven't made an example yet that documents this feature. Would that be interesting for you?

Best,
Uwe

**Comment by romainGuiet** [comment]
Hi Uwe,

thank you for your prompt answer!

> What do you mean by "sensitive to scaling"?

On my original image, the nuclei are 20-60 pixels large amd some are fragmented. So I had a look on the training dataset (I have to admit now that I only looked at the first image) and saw that they were more in the range 10-30 pixels. So I tried to dowsample my image with different value, please look at this [scaling_results.pdf](https://github.com/mpicbg-csbd/stardist/files/3321161/scaling_results.pdf) with the ROIs obtained for 2 images. I think the best results are obtained with the downsampling 0.75 and 0.5.

 > I have a basic StarDist Fiji plugin working, but can't promise at the moment when it'll be done.

 Sounds promising! Looking foward for a beta!

For the moment I modified your jupyter notebook to export the labels as tifs and I transformed the labels image into ROIs in Fiji so :
>the possibility to export the StarDist projections as ImageJ1 ROIs. The Python code is done, but I haven't made an example yet that documents this feature.

could save me 1 step.

Both solution are fine for prototyping but I would like to include this awesome segmentation in a workflow to process large images. Ideally our user would do it themselves, and while they are now familiar with ImageJ/Fiji, using jupyter notebooks would still require some trainings.

Best,

Romain

**Comment by uschmidt83** [comment]
Hi Romain,
> I think the best results are obtained with the downsampling 0.75 and 0.5.
> 
The provided training data contain nuclei with a variety of sizes. Of course, you could train the model only with specific sizes, thereby making a model for your specific nuclei sizes.
> For the moment I modified your jupyter notebook to export the labels as tifs and I transformed the labels image into ROIs in Fiji
> 
This will work for non-overlapping nuclei.
> the possibility to export the StarDist projections as ImageJ1 ROIs. The Python code is done, but I haven't made an example yet that documents this feature.
> 
> could save me 1 step.
> 
This allows you to export overlapping ROIs. 
> Both solution are fine for prototyping but I would like to include this awesome segmentation in a workflow to process large images.
> 
What do you mean by large images?

Best,
Uwe

**Comment by romainGuiet** [comment]
Hi Uwe,

and sorry for the long silence. -_-'

> What do you mean by large images?
Please find below an example image of what our user would have to process , ie 9000 x 5000 pixels (here as jpg due to upload limit).
![Fused-w1](https://user-images.githubusercontent.com/8309560/60800577-bddaf600-a175-11e9-9055-2d9e40d7aaf2.jpg) 

Best,

Romain

**Comment by bharath5673** [comment]
please help even for e too.. i need to save induvidual ROIS as induvidual files in stardist

**Comment by uschmidt83** [comment]
Hi @bharath5673,

please ask question like this in the [image.sc forum](https://forum.image.sc/).
(Use the tag `stardist` if your question is StarDist-related.)

Best,
Uwe
